tests/conftest.py

```diff
@@ -406,7 +431,20 @@ def pytest_terminal_summary(terminalreporter, exitstatus, config):
         if summary_path := os.environ.get("GITHUB_STEP_SUMMARY"):
             summary_path = Path(summary_path).resolve()
             with summary_path.open("a") as f:
-                f.write("## Failed tests\n")
+                # Write cross-version matrix info if available
+                if matrix_json := os.environ.get("CROSS_VERSION_MATRIX"):
+                    try:
+                        matrix = json.loads(matrix_json)
+                        matrix.pop("run", None)
+                        formatted_yaml = _dump_yaml(matrix)
+                        f.write("\n### Cross-Version Test Configuration\n\n")
+                        f.write("```yaml\n")
+                        f.write(formatted_yaml)
+                        f.write("```\n\n")
+                    except json.JSONDecodeError:
+                        pass
+
+                f.write("### Failed Tests\n\n")
```

Comment:
The summary looks like this:

<img width="1339" height="484" alt="image" src="https://github.com/user-attachments/assets/b0a123df-722a-495b-ad78-c4bb97da0bc1" />